src/xenia/kernel/xam/xam_content.cc

```diff
@@ -133,10 +133,11 @@ dword_result_t XamContentGetDeviceData(
 
   device_data.Zero();
   const auto& device_info = dummy_device_info_;
-  device_data->device_id = device_info.device_id;
-  device_data->unknown = device_id & 0xFFFF;  // Fake it.
-  device_data->total_bytes = device_info.total_bytes;
-  device_data->free_bytes = device_info.free_bytes;
+  // byteswap the device info
+  device_data->device_id = xe::byte_swap(device_info.device_id);
```

Comment:
This shouldn't be necessary, `X_CONTENT_DEVICE_DATA` has all of its fields wrapped in `xe::be`?

Reply:
testing proved otherwise? I removed the obscene amount of debug prints that led me to find this. We're swapping it in XamContentCreateDeviceEnumerator and games that used that seemed to be working, I followed suit when I noticed this function was failing to work properly and voilà.

Reply:
The code that you're referring to is copying data by pointers to struct fields, not simply assigning swapped values to the struct. Thus the swaps.

https://github.com/xenia-project/xenia/blob/4f5c313fe39350c7abd86150bf0558e23b5d98d1/src/xenia/kernel/xam/xam_content.cc#L222-L226

Reply:
I'm not saying your change is a bad one, just that the byte swap here seems out of place / wrong.

Reply:
Actually, I think it is wrong to byteswap here you were correct to doubt it, I made a false correlation here and didn't test enough.